packages/headless/src/features/product-recommendations/product-recommendations-analytics.actions.ts

```diff
@@ -13,3 +25,95 @@ export const logProductRecommendations = makeAnalyticsAction(
       state as StateNeededByProductRecommendationsAnalyticsProvider
     )
 );
+
+const partialRecommendationInformation = (
+  result: ProductRecommendation,
+  state: Partial<ProductRecommendationsAppState>
+): PartialDocumentInformation => {
+  const resultIndex =
+    state.productRecommendations?.recommendations.findIndex(
+      ({permanentid}) => result.permanentid === permanentid
+    ) || 0;
+
+  return buildPartialDocumentInformation(result, resultIndex);
+};
+
+function buildPartialDocumentInformation(
+  productRecommendation: ProductRecommendation,
+  resultIndex: number
+): PartialDocumentInformation {
+  return {
+    collectionName: '',
```

Comment:
If some of them are going to be empty anyway, are we adding them because they are needed by `coveo.analytics` ?

Reply:
Yes, validation is expected all these fields.